### PR TITLE
Store: WooCommerce: Add site descriptor field to Stripe settings

### LINKED ISSUE #16659: Store: Stripe Statement Descriptor Not Editable
When setting up the Stripe payment option in the calypso store, the "Statement Descriptor" is not editable via the user interface:

payment_settings_ _care_for_animals_ _wordpress_com

After placing a test order, I had assumed the "business details" from Stripe would have been used in the transaction, but the "Statement Descriptor" was still used from woo:

payment_of__5_00_from_timmy_crawford_-_timmydcrawford_gmail_com_-_gmail

In order to change this setting, I had to visit wp-admin which for a completely new user is not very intuitive.

### PR DESCRIPTION
**Pull request by allendav**
Fixes #16659 

To test:
* Go to http://calypso.localhost:3000/store/{site}
* Go to Settings > Payments > Stripe > Manage
* Make sure the descriptor is visible, editable, and that clearing it generates a suggested value based on the site's domain name
* Ensure you are not allowed to type more than 22 characters in the field
* Ensure that if you change the value, close the modal, and hit save, that the changes are persisted to the site

screen shot 2017-07-31 at 3 12 30 pm

### CODE REVIEW

**Comment by allendav** [comment]
cc @kellychoffman or @jameskoster for design fu

File: client/extensions/woocommerce/app/settings/payments/payment-method-stripe.js [hunk 1]
```diff
@@ -152,11 +169,26 @@ class PaymentMethodStripe extends Component {
 							value="no"
 							checked={ 'no' === method.settings.capture.value }
 							onChange={ this.onEditFieldHandler } />
-						<span>{ translate( 'Authorize the customers credit card but charge manually' ) }</span>
+						<span>{ translate( 'Authorize the customer\'s credit card but charge manually' ) }</span>
+					</FormLabel>
+				</FormFieldset>
+				<FormFieldset>
+					<FormLabel>
+						{ translate( 'Descriptor' ) }
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/settings/payments/payment-method-stripe.js]: Hi! I've found a possible matching string that has already been translated **44** times:
`translate( 'Descripton' )` *ES Score: 9*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/app/settings/payments/payment-method-stripe.js [hunk 2]
```diff
@@ -36,9 +37,19 @@ class PaymentMethodStripe extends Component {
 		onCancel: PropTypes.func.isRequired,
 		onEditField: PropTypes.func.isRequired,
 		onDone: PropTypes.func.isRequired,
+		site: PropTypes.shape( {
+			title: PropTypes.string,
+		} ),
 	};
 
 	onEditFieldHandler = ( e ) => {
+		// Limit the statement descriptor field to 22 characters
```
Comment (budzanowski) [inline client/extensions/woocommerce/app/settings/payments/payment-method-stripe.js]: Why 22?

File: client/extensions/woocommerce/app/settings/payments/payment-method-stripe.js [hunk 3]
```diff
@@ -36,9 +37,19 @@ class PaymentMethodStripe extends Component {
...
+		if ( e.target && e.target.name && 'statement_descriptor' === e.target.name ) {
```
Comment (budzanowski) [inline client/extensions/woocommerce/app/settings/payments/payment-method-stripe.js]: I think that you can drop the middle check `e.target.name`. The next one will still work even if name is not defined.

**Comment by budzanowski** [comment]
Left small nit - nothing blocking. Code looks good. Ship it!

File: client/extensions/woocommerce/app/settings/payments/payment-method-stripe.js (hunk 2 above)
Comment (allendav) [inline client/extensions/woocommerce/app/settings/payments/payment-method-stripe.js]: Stripe's limit per https://stripe.com/docs/connect/statement-descriptors

> Note that although we enforce a maximum of 22 characters, most banks truncate this information or display it inconsistently. Some banks may not display any descriptor information at all.

Comment (allendav) [inline client/extensions/woocommerce/app/settings/payments/payment-method-stripe.js]: I will add this to the comment

File: client/extensions/woocommerce/app/settings/payments/payment-method-stripe.js (hunk 3 above)
Comment (allendav) [inline client/extensions/woocommerce/app/settings/payments/payment-method-stripe.js]: Good idea. Will do. I tend to over prescribe these things

Comment (allendav) [inline client/extensions/woocommerce/app/settings/payments/payment-method-stripe.js]: Fixed in https://github.com/Automattic/wp-calypso/pull/16774/commits/72de550e887a557ad2e4610c6e5ee1c6da3c9c3d

File: client/extensions/woocommerce/app/settings/payments/payment-method-stripe.js (hunk 2 above)
Comment (allendav) [inline client/extensions/woocommerce/app/settings/payments/payment-method-stripe.js]: Fixed in https://github.com/Automattic/wp-calypso/pull/16774/commits/72de550e887a557ad2e4610c6e5ee1c6da3c9c3d

**Comment by fjalestad** [comment]
Would you consider to att a filter to the stripe_description here:
`
$post_data['description'] = sprintf( __( '%1$s - Order %2$s', 'woocommerce-gateway-stripe' ), $this->statement_descriptor, $order->get_order_number() );
`
like you have for meta_data?

This would allow people to customize the description as the wish.

**Pull request merged**